src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessedEnvelopeNotificationHandler.java

```diff
@@ -46,8 +48,8 @@ public ProcessedEnvelopeNotificationHandler(
     @Override
     public CompletableFuture<Void> onMessageAsync(IMessage message) {
         return CompletableFuture
-            .supplyAsync(() -> tryProcessMessage(message))
-            .thenAcceptAsync(processingResult -> tryFinaliseMessage(message, processingResult))
+            .supplyAsync(() -> tryProcessMessage(message), executor)
```

Comment:
still want to go with "full blown" `tryProcess`?

Reply:
There's a lot of error handling logic that in my opinion would be best not to expose at this level.

Reply:
I've added a comment in handleAsync to make it clear what purpose it serves there

Reply:
which is why single point of entry to deal with exceptions is the exception handler. where you can catch everything singularly despite how origin works. so, in short, processing can be left alone with just doing what it suppose to do. any exception? - handle in exception handler

Reply:
I completely disagree. It matters which exception is thrown at which point. We have to catch exceptions in the first step, in order to provide the second step with the right result (so that it knows what to do with the message). And when we fail to do some specific thing, it's much better to see an error saying we've failed to do just that (with details that are available in this context) than that the whole task failed.

When getting rid of context by centralising exception handling we're likely to overcomplicate our logic. Single exception handlers aren't always the right thing to do, unless they handle some general failures, just like the one in the main method.